### ISSUE TITLE
[BUG] Training Skill Estimation thinks everyone pops next week (v6)

### ISSUE THREAD

**Issue by Synthespian** [akasolace/HO#1541]
**Describe the bug**
Instead of calculating the previous trainings from the last skill pops, the estimation page (even after a subskill recalculation) has the entire team calculated as improving in skill level the following week UNLESS:
* they have gone up a skill level since using v6
* been recently bought 
(these represent the two not showing as described in the screenshot)

**To Reproduce**
1. Go to 'Training > Prediction'
2. Review players and see issue
3. Go to 'Training > Training'
4. Click 'Subskill recalculation'
5. Go to 'Training > Prediction'
5. See error still persists

**Screenshots**
![image](https://user-images.githubusercontent.com/48737160/175489051-2d37a9bd-6e46-42d6-93c6-e1c940deb710.png)

**Platform information:**
 - OS: Windows
 - Version 19043.1766

**Additional context**
Using JRE included standalone version

**Comment by wsbrenk** [comment]
@Synthespian i can not reproduce this with my team. if you could provide your database backup i will try to debug your situation.

the only idea I have is that the calculation starts on a corrupt record of players.

**Comment by Synthespian** [comment]
All good.  Happy to send. It's correct on 5, from the same starting point (I just copied the db over).
![image](https://user-images.githubusercontent.com/48737160/175771255-1d6ce94f-15f7-4040-81d2-70b6e560cd69.png)

db file size is too big for here.  Thoughts for sending?

**Comment by wsbrenk** [comment]
@Synthespian Please run the skill recalculation and check if the log file includes some helpful messages.

**Comment by wsbrenk** [comment]
@Synthespian the backup zip file is really bigger than 100MByte? 

Did you try to run File->Database->Database cleanup 
and restarting HO (to backup the cleaned database)?

**Comment by Synthespian** [comment]
I naturally have already run the recalculation, so would already be in the log file.
And in trying to upload, it said the max was 25 or 30 and my backup is 35.

Not at home at the moment, but will check again when I'm back.

**Comment by wsbrenk** [comment]
@Synthespian you're right, i fear. 100Mbyte is another github file limit. attachments of issues are limited to 25Mbyte.

**Comment by Synthespian** [comment]
backup: https://drive.google.com/file/d/1zV4brWmHVMlzb_vERqTj_kST6wct219D/view?usp=sharing
log: https://drive.google.com/file/d/1zushT1dX0lBqmkOFItrqhNvY1v2GfQIb/view?usp=sharing
output.log https://drive.google.com/file/d/1zktW7gFOi4-OSsfQ9saKD_l4iRjS4c5m/view?usp=sharing

**Comment by wsbrenk** [comment]
debugging is in progress.

After adjusting wrong defined past trainings the recalculation worked better:
![image](https://user-images.githubusercontent.com/11821502/176282020-34229898-568f-4970-8020-acefd53b7a9e.png)

i will try to fix that - not defined training should result in no additional sub skills

**Comment by wsbrenk** [comment]
got it

**Comment by Synthespian** [comment]
Makes sense.
Interesting tho that despite being a backup from v5, it's got different entries in the past trainings (which I hadn't changed I don't believe - could have in a random fit)
![image](https://user-images.githubusercontent.com/48737160/176466659-ba0374c5-96cf-48c7-a889-2be6b6d91bb7.png)

And that wasn't therefore causing the same failure in v5.
I've updated the one remaining undefined in v6 and recalculated.

No update.
Filling in all the red items to blue and numbers to same 100/10/8/10 then it worked.
